src/main/java/uk/gov/companieshouse/filinghistory/consumer/delta/FilingHistoryDeltaProcessor.java

```diff
@@ -0,0 +1,40 @@
+package uk.gov.companieshouse.filinghistory.consumer.delta;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.delta.FilingHistory;
+import uk.gov.companieshouse.api.delta.FilingHistoryDelta;
+import uk.gov.companieshouse.api.filinghistory.InternalFilingHistoryApi;
+import uk.gov.companieshouse.filinghistory.consumer.delta.transformrules.TransformerService;
+
+@Component
+public class FilingHistoryDeltaProcessor {
+
+    private final TransactionKindService kindService;
+    private final PreTransformMapper preTransformMapper;
+    private final TransformerService transformerService;
+    private final InternalFilingHistoryApiMapper internalFilingHistoryApiMapper;
+
+    public FilingHistoryDeltaProcessor(TransactionKindService kindService, PreTransformMapper preTransformMapper, TransformerService transformerService, InternalFilingHistoryApiMapper internalFilingHistoryApiMapper) {
+        this.kindService = kindService;
+        this.preTransformMapper = preTransformMapper;
+        this.transformerService = transformerService;
+        this.internalFilingHistoryApiMapper = internalFilingHistoryApiMapper;
+    }
+
+    public InternalFilingHistoryApi processDelta(FilingHistoryDelta delta, final String updatedBy) {
+        final FilingHistory filingHistory = delta.getFilingHistory().getFirst();
+        final JsonNode transformedJsonNode = transformerService.transform(
+                preTransformMapper.mapDeltaToObjectNode(filingHistory));
+
+        return internalFilingHistoryApiMapper.mapJsonNodeToInternalFilingHistoryApi(
+                transformedJsonNode, buildKindResult(filingHistory), delta.getDeltaAt(), updatedBy);
+    }
+
+    private TransactionKindResult buildKindResult(final FilingHistory filingHistory) {
+        return kindService.encodeIdByTransactionKind(
```

Comment:
Private methods aren't testable directly so I'd like for the kind service to be used within processDelta so its clear which dependencies are being used in the method. I'd be happy for you to make it a private static method which takes the FilingHistory and returns the TransactionKindCriteria though.